test/unit/atomic/src/AtomicTest.cpp

```diff
@@ -425,8 +425,10 @@ class AtomicTestKernel
 public:
     //-----------------------------------------------------------------------------
     ALPAKA_NO_HOST_ACC_WARNING
+    template<
+        typename TTAcc>
     ALPAKA_FN_ACC auto operator()(
-        TAcc const & acc,
+        TTAcc const & acc,
```

Comment:
Why?

Reply:
Kernel `operator()` must itself be a template with respect to `acc`, so the backend can pass the thread context.

The class already has `TAcc` as template parameter. This patch adds a template parameter `TTAcc` for the `operator()`, named this way to not trigger `-Wshadow`.

Reply:
I do not get it. If `operator()` needs to have the template parameter, then `AtomicTestKernel` should not have it. Please remove it from the class then. Or does the class support to be instantiated on two different accelerators?

Reply:
All tests are explicitly templated on the accelerator. I did not know what implications removing the template here, because this could go into catch2. I can have a try.

Reply:
The template param `TAcc` cannot be removed from the class `AtomicTestKernel` because it is specialized on this param for the HIP backend. However, technically, after merging `CtxThreadOacc` and `AccOacc` the template param on the `operator()` is no longer required, but not having it does not appear clean. Still, I will remove it from this PR.